torchtitan/experiments/deepseek_v3/moe_kernels.py

```diff
@@ -0,0 +1,261 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import torch
+import triton
+import triton.language as tl
+
+
+__all__ = ["generate_permute_indices"]
+
+
+# parallelized kernel
+@triton.jit
+def _fill_indices_kernel(
+    tokens_per_expert_group_ptr,
+    start_index_values_ptr,
+    write_offsets_ptr,
+    output_ptr,
+    experts_per_rank: tl.constexpr,
+    num_ranks: tl.constexpr,
+    total_experts: tl.constexpr,
+    max_blocks: tl.constexpr,
+    experts_per_block: tl.constexpr,
```

Comment:
no need for these three arguments?

Reply:
with corrected looping, yes no need and have removed.